source/Calamari.AzureAppService/Behaviors/AzureAppServiceZipDeployBehaviour.cs

```diff
@@ -144,7 +151,15 @@ public async Task Execute(RunningDeployment context)
             Log.Verbose($"Using deployment endpoint '{publishingProfile.PublishUrl}' from publishing profile.");
 
             Log.Info($"Uploading package to {targetSite.SiteAndSlot}");
-            await UploadZipAsync(publishingProfile, uploadPath, targetSite.ScmSiteAndSlot);
+
+            if (FeatureToggle.AsynchronousAzureZipDeployFeatureToggle.IsEnabled(context.Variables))
```

Comment:
I may be wrong where @APErebus, but from reading the [doco](https://learn.microsoft.com/en-us/azure/azure-functions/deployment-zip-push#rest), this method seems to only be supported for `/api/zipdeploy` which is the [ZipPackageProvider](https://github.com/OctopusDeploy/Calamari/blob/6981c41c7ffad05dfc8ccf95c8f9b20043b04618/source/Calamari.AzureAppService/ArchivePackagProvider/ZipPackageProvider.cs#L11), so I think we need to restrict this path only if `Archive is ZipPackageProvider`?

Reply:
Hrm ok. This isn't something I was aware of. Thanks for the heads up. Let me check further 👍

Reply:
@johnsimons Zip & Nuget both use the `/api/zipdeploy` endpoint, so I've set them as both supporting async deployment. I added a new test to cover the nuget async deployment and it works 👍